dev_tools/scripts/create_new_algorithm.sh

```diff
@@ -24,9 +24,9 @@ function print_help {
 function list_sources {
     for type in solver preconditioner matrix factorization
     do
-        for i in $(ls $GINKGO_ROOT_DIR/core/$type/*.cpp)
+        for i in "${GINKGO_ROOT_DIR}/core/${type}/*.cpp";
```

Comment:
This doesn't work with my bash on my computer because the `*` is quoted
```suggestion
        for i in "${GINKGO_ROOT_DIR}/core/${type}/"*.cpp;
```

Reply:
weird, I suggested this change because it works on my system, I thought globs are expanded inside `""` environments, just not in `''`

Reply:
I have `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)` and it somehow definitely doesn't work here.

Reply:
Ah, I see what is going wrong: The glob is being expanded, but into a single string:
`for f in "*"; do echo $f; done` lists all files in a single line, while removing the quotes gets them in separate lines. Then this needs to be changed, yes.